proof/refine/KHeap_R.thy

```diff
@@ -9,6 +11,59 @@ imports
   ArchMachine_R
 begin
 
+arch_requalify_facts
+  valid_arch_obj'_valid
+
+lemmas [wp] = valid_arch_obj'_valid
+
+(* FIXME: this should go somewhere in spec *)
+translations
+  (type) "'a kernel" <=(type) "kernel_state \<Rightarrow> ('a \<times> kernel_state) set \<times> bool"
+
+declare fromPPtr_def[simp] (* FIXME arch-split: move this somewhere high up *)
```

Comment:
KHeap is probably high enough, tbh. I think we can remove the FIXME from this one.

Reply:
Will do

Reply:
done, will show up in next push

Reply:
done